lib/map/security_token/service.rb

```diff
@@ -57,17 +61,25 @@ def parse_and_raise_error(e, icn, application)
 
       def parse_response(response, application, icn)
         response_body = response.body
+        validate_map_token(response_body['access_token'])
 
         {
           access_token: response_body['access_token'],
           expiration: Time.zone.now + response_body['expires_in']
         }
+      rescue JWT::DecodeError => e
+        raise e
       rescue => e
         message = "#{config.logging_prefix} token failed, response unknown"
         Rails.logger.error(message, application:, icn:)
         raise e, "#{message}, application: #{application}, icn: #{icn}"
       end
 
+      def validate_map_token(encoded_token)
+        jwk = config.public_jwks.first
```

Comment:
We should decode with all the available JWKs, not just the first. Otherwise this will break when the key is rotated